### PR TITLE
Upgrade Composer usage

### PR DESCRIPTION
**Pull request by szepeviktor**
## Describe the PR

- Travis images include self-upgrade
- Require two packages at once
- Always validate config file

## Ready?

- [x] The lock file needs synchronization

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,10 +15,9 @@ before_install:
 
 install:
   # Install composer packages, will also trigger dump-autoload
-  - composer self-update
-  - composer require phpunit/phpunit ^7
-  - composer require friendsofphp/php-cs-fixer
-  - composer install --no-interaction --ignore-platform-reqs
+  - composer validate
```
Comment (lukasbestle) [inline .travis.yml:18]: Why did you move the `composer validate` command to the `install` step? It's part of the actual test suite, so I think it should run later as a `script`.

I agree on the other changes.

**Review by lukasbestle — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (szepeviktor) [reply .travis.yml:18]: I think and invalid composer config is a road block, so it should be number 1.

Comment (lukasbestle) [reply .travis.yml:18]: That's right. Our Composer config is rarely completely invalid, but when it is it should fail immediately.

**Review by lukasbestle — DISMISSED** [review]

**Comment by lukasbestle** [comment]
Hm, what I don't like about `composer validate` being part of the install steps is that the output gets collapsed if the step succeeded (= warnings aren't visible). But well, it all has pros and cons.

**Comment by szepeviktor** [comment]
add `--strict` and it will: _Return a non-zero exit code for warnings as well as errors_

**Comment by lukasbestle** [comment]
We don't want that either. As you can see, we currently have warnings and we deliberately want to ignore them. But we want to see when new warnings get output (for example when the Composer team changes anything in the config format) so that we can review the warnings manually.

**Comment by szepeviktor** [comment]
Please clean up all warnings. Benefit is hidden inside each of them, although it looks like difficulty, something you don't want.

**Comment by lukasbestle** [comment]
We can't do that – while we could switch to less specific version constraints for the dependencies (and indeed probably should @bastianallgeier), the other warning about the `version` field in `composer.json` can't be fixed as we rely on that internally.

**Review by bastianallgeier — APPROVED** [review]

**Comment by szepeviktor** [comment]
I'm thankful for becoming a Kirby contributor.

**Pull request merged**